File: problems/101/solution.go
package main
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isSymmetric(root *TreeNode) bool {
    return traverse(root, root)
}

func traverse(left, right *TreeNode) bool {
    if left == nil && right == nil {
        return true
    }
    
    if left == nil || right == nil {
        return false
    }
    
    return left.Val == right.Val && traverse(left.Left, right.Right) && traverse(left.Right, right.Left)
}

File: problems/101/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        q = deque([(root, root)])
        
        while q:
            left, right = q.popleft()
            
            if not left and not right:
                continue
            
            if not left or not right:
                return False
            
            if left.val != right.val:
                return False
        
            q.append((left.left, right.right))
            q.append((left.right, right.left))
        
        return True
                